spec/rester_spec.rb

```diff
@@ -128,4 +130,113 @@
       end
     end # without circuit_breaker_enabled
   end # ::connect
+
+  describe '::service_name' do
+    subject { Rester.service_name }
+
+    it 'should default to the Rails application name' do
+      is_expected.to eq "Dummy"
+    end
+
+    context 'with service_name set' do
+      before { Rester.service_name = "New Consumer Name" }
+      it { is_expected.to eq "New Consumer Name" }
+    end # with service_name set
+  end # ::service_name
+
+  describe '::begin_request' do
+    subject { Rester.request_info }
+    before { Rester.begin_request }
+    it { is_expected.to eq({}) }
+  end # ::begin_request
+
+  describe '::end_request' do
+    subject { Rester.request_info }
+    before { Rester.end_request }
+    it { is_expected.to eq nil }
+  end # ::end_request
+
+  describe '::request_info=' do
+    subject { Rester.request_info }
+    before { Rester.request_info = request_info }
+    let(:request_info) {  { correlation_id: id } }
+    let(:id) { SecureRandom.uuid }
+
+    context 'with single thread' do
+      context 'with new request_info' do
+        it 'should set the request info value' do
+          expect(subject[:correlation_id]).to eq id
+        end
+      end # with new request_info
+
+      context 'with nil request_info' do
+        it 'should delete the request_info for this thread' do
+          expect(subject[:correlation_id]).to eq(id)
+          subject = nil
+          expect(subject).to eq nil
+        end
+      end
+    end # with single thread
+
+    context 'with multiple threads' do
+      before { new_thread.join }
+      let(:new_thread) {
+        Thread.new { Rester.request_info = { correlation_id: new_id } }
+      }
+      let(:new_id) { SecureRandom.uuid }
+      subject { Rester.instance_variable_get(:@_request_infos) }
+
+      context 'with new id' do
+        it 'should set the correlation id for both threads' do
+          expect(subject[Thread.current.object_id]).to eq(correlation_id: id)
+          expect(subject[new_thread.object_id]).to eq(correlation_id: new_id)
+        end
+      end # with new id
+
+      context 'with nil id' do
+        it 'should delete the key for the current thread' do
+          expect(subject[Thread.current.object_id]).to eq(correlation_id: id)
+          expect(subject[new_thread.object_id]).to eq(correlation_id: new_id)
+          Rester.request_info = nil
+          expect(subject[Thread.current.object_id]).to eq nil
+          expect(subject[new_thread.object_id]).to eq(correlation_id: new_id)
+        end
+      end # with nil id
```

Comment:
What about instead of accessing the private instance variable `@_request_infos`, doing something like:

```ruby
let(:new_thread_info) { { correlation_id: new_id } }

let(:new_thread) do
  Thread.new do
    Rester.request_info = new_thread_info
    expect(Rester.request_info).to eq new_thread_info
  end
end

before { new_thread.join }
it { expect(Rester.request_info).to eq request_info }
```

This tests that even though the thread runs after the original `Rester.request_info = request_info`, `Rester.request_info` still equals `request_info` even though the thread modifies its value.